domain-models-interface-extensions/src/main/java/org/folio/rest/tools/utils/ClassPath.java

```diff
@@ -0,0 +1,566 @@
+     * File Specification</a>. If {@code manifest} is null, it means the jar file has no manifest,
+     * and an empty set will be returned.
+     */
+    static Set<File> getClassPathFromManifest(File jarFile, Manifest manifest) {
+      if (manifest == null) {
+        return Collections.emptySet();
+      }
+
+      Set<File> files = new LinkedHashSet<>();
+
+      String classpathAttribute = manifest.getMainAttributes().getValue(Attributes.Name.CLASS_PATH.toString());
+      if (classpathAttribute != null) {
+        for (String path : StringUtils.splitByWholeSeparator(classpathAttribute, null)) {
+          URL url;
+          try {
+            url = getClassPathEntry(jarFile, path);
+          } catch (MalformedURLException e) {
+            // Ignore bad entry
+            log.warn("Invalid Class-Path entry: " + path);
+            continue;
+          }
+          if (url.getProtocol().equals("file")) {
+            files.add(toFile(url));
+          }
+        }
+      }
+
+      return Set.copyOf(files);
+    }
+
+    static Map<File, ClassLoader> getClassPathEntries(ClassLoader classloader) {
+      Map<File, ClassLoader> entries = new LinkedHashMap<>();
+
+      // Search parent first, since it's the order ClassLoader#loadClass() uses.
+      ClassLoader parent = classloader.getParent();
+      if (parent != null) {
+        entries.putAll(getClassPathEntries(parent));
+      }
+
+      for (URL url : getClassLoaderUrls(classloader)) {
+        if (url.getProtocol().equals("file")) {
+          File file = toFile(url);
+
+          if (!entries.containsKey(file)) {
+            entries.put(file, classloader);
+          }
+        }
+      }
+
+      return Map.copyOf(entries);
+    }
+
+    private static List<URL> getClassLoaderUrls(ClassLoader classloader) {
+      if (classloader instanceof URLClassLoader) {
+        return List.of(((URLClassLoader) classloader).getURLs());
+      }
+      if (classloader.equals(ClassLoader.getSystemClassLoader())) {
+        return parseJavaClassPath();
+      }
+      return Collections.emptyList();
+    }
+
+    /**
+     * Returns the URLs in the class path specified by the {@code java.class.path} {@linkplain
+     * System#getProperty system property}.
+     */
+    static List<URL> parseJavaClassPath() {
+      List<URL> urls = new LinkedList<>();
+
+      String pathSeparator = SystemUtils.PATH_SEPARATOR;
+      String javaClassPath = SystemUtils.JAVA_CLASS_PATH;
+
+      for (String entry : StringUtils.split(javaClassPath, pathSeparator)) {
+        try {
+          try {
+            urls.add(new File(entry).toURI().toURL());
+          } catch (SecurityException e) { // File.toURI checks to see if the file is a directory
+            urls.add(new URL("file", null, new File(entry).getAbsolutePath()));
+          }
+        } catch (MalformedURLException e) {
+          log.warn("Malformed classpath entry: " + entry, e);
+        }
+      }
+
+      return List.copyOf(urls);
+    }
```

Comment:
ClassPathTest.testGetClassPathUrls fails.

Reply:
I have two tests failing.. When I run them individually they all succeed. SO not isolated. I'm inclined to drop the test unless somebody has an easy fix to provoke *some* order that works.

Reply:
I would rather fix the implementation bugs.

Reply:
Implementation bugs fixed.